### PR TITLE
Borers now transfer, drop, and delete accordingly during transformation or body destruction

### LINKED ISSUE #6870: Borer host got borged, borer stil alive in null loc hell
(WEB REPORT BY: pseudonym REMOTE: 158.69.55.2:7777)

> Revision (Should be above if you're viewing this from ingame!)
> dee5cb7b250b0f88e0fde862b2bb8472d38e485c 
> 
> General description of the issue
> Was playing a borer. Was inside a host that got borged. I was still alive. I hit 'Abandon Host', and it succeeded, but I had no location
> 
> What you expected to happen
> Would expect that the borer would die, or that it would be ejected on the host's gibbing/borging
> 
> What actually happened
> Null loc hell
> 
> Steps to reproduce if possible
> Be a borer
> Get a host
> Have host be borged
> Leave host, check location.

### LINKED ISSUE #7359: Borer inside a soul sharded person will be sent to nullspace

### PR DESCRIPTION
**Pull request by Intigracy**
fixes #6870
fixes #7359

### CODE REVIEW

File: code/modules/mob/living/carbon/human/death.dm [hunk 1]
```diff
@@ -29,10 +24,7 @@
 	icon = null
 	invisibility = 101
 
-	//If we have brain worms, dump 'em.
-	var/mob/living/simple_animal/borer/B=has_brain_worms()
-	if(B)
-		B.detach()
+	dropBorers()
```
Comment (Intigracy) [inline code/modules/mob/living/carbon/human/death.dm]: Need to change to dropBorers(1)

**Comment by Intigracy** [comment]
This is ready unless anyone has an issues.

File: code/game/dna/genes/monkey.dm [hunk 2]
```diff
@@ -39,6 +38,7 @@
 	else
 		for(var/obj/item/W in (H.contents))
 			H.drop_from_inventory(W)
+		H.dropBorers()
```
Comment (clusterfack) [inline code/game/dna/genes/monkey.dm]: gibbed arg

**Comment by clusterfack** [comment]
Also dropping the borer instead of transferring it is an odd choice

**Comment by Intigracy** [comment]
I figure if it's changing the mob it's probably changing the brain too, which is where the borer is attached.

In the case of lings it's for balance. If you're going to monkey, your borer is gonna pop out and if you want him you'll have to wait for him.

**Comment by clusterfack** [comment]
Well it seems like it should be handled like implants to me

**Comment by Intigracy** [comment]
I could make it so that if the mob type isn't changing (like evolving slimes/aliens) that it keeps the borer.

**Comment by Intigracy** [comment]
@clusterfack @N3X15 This is no longer DNM, just needs final decision.

File: code/game/verbs/suicide.dm [hunk 3]
```diff
@@ -119,20 +119,20 @@
...
 
 	if(suiciding)
-		to_chat(src, "<span class='warning'>You're already committing suicide! Be patient!</span>")
+		to_chat(src, "<span class='warning'>You're already committing suicide! Be patient!</span>")
 		return
 
 	var/mob/living/simple_animal/borer/B=has_brain_worms()
 	if (B && B.controlling) // Borer
-		to_chat(src, "You cannot suicide as your host, you dick.")
+		to_chat(src, "Your borer prevents you from committing suicide. You dick.")
```
Comment (clusterfack) [inline code/game/verbs/suicide.dm]: I think this changes the original intention

File: code/modules/mob/living/carbon/brain/brain.dm [hunk 4]
```diff
@@ -24,7 +24,7 @@
 	if(key)				//If there is a mob connected to this thing. Have to check key twice to avoid false death reporting.
 		if(stat!=DEAD)	//If not dead.
 			death(1)	//Brains can die again. AND THEY SHOULD AHA HA HA HA HA HA
-		ghostize()		//Ghostize checks for key so nothing else is necessary.
```
Comment (clusterfack) [inline code/modules/mob/living/carbon/brain/brain.dm]: You can remove this anyway, mob/Destroy() already calls ghostize(0)

File: code/game/verbs/suicide.dm (hunk 3 above)
Comment (Intigracy) [inline code/game/verbs/suicide.dm]: Oh, I see now. This is the borer controlling the human trying to suicide.

File: code/modules/mob/living/carbon/brain/brain.dm (hunk 4 above)
Comment (Intigracy) [inline code/modules/mob/living/carbon/brain/brain.dm:27]: done

**Pull request merged**